pkgs/servers/klipper/klipper-flash.nix

```diff
@@ -0,0 +1,41 @@
+{ lib
+, writeShellApplication
+, gnumake
+, pkgsCross
+, klipper
+, klipper-firmware
+, python2
+, avrdude
+, stm32flash
+, mcu ? "mcu"
+, flashDevice ? "/dev/null"
+, firmwareConfig ? ./simulator.cfg
+}:
+let
+  isNotSupported = with builtins; isNull (match ''^.*CONFIG_BOARD_DIRECTORY="(avr|stm32|lpc176x)".*$'' (readFile firmwareConfig));
+  isNotStm = with builtins; isNull (match ''^.*CONFIG_BOARD_DIRECTORY="(stm32)".*$'' (readFile firmwareConfig));
```

Comment:
I think this is better done with a single var: `boardArch`. You can then just check whether `boardArch` is in `supportedArches`.

You then can also make `runtimeInputs` conditional on the `boardArch`.

Reply:
Done.